doc/content/devices/confirmed-downlinks-behavior/_index.md

```diff
@@ -0,0 +1,25 @@
+---
+title: "Confirmed Downlinks Behavior"
+description: ""
+weight: 8
+---
+
+This section describes differences in behavior of class A, B and C devices when scheduling confirmed downlink messages.
+
+<!--more-->
+
+Let's consider scheduling a confirmed downlink message from the Network Server. The end device that receives this downlink is requested to send a confirmation upon its reception.
+
+To confirm a reception of the above mentioned downlink, a confirmation flag is attached to device's next uplink message. The confirmation flag is part of the standard LoRaWAN frame header. If the flag is an NACK, the end device indicates that it has not previously received a downlink message that required acknowledgement. If the flag is a ACK, the end device indicates that it wants to provide the acknowledgement for the previously received downlink message that requested it. This applies to all [device classes](https://www.thethingsnetwork.org/docs/lorawan/classes/).
+
+In general, upon receiving the next uplink message with an ACK flag, the Network Server decrypts the uplink message and generates the `downlink ack` message. The Application Server receives the `downlink ack` message and  forwards it further to the integrations.
+
+Similar to that, if the next uplink message with a NACK flag is received, the Network Server decrypts the uplink, generates the `downlink nack` message and sends it to the Application Server. The Application Server then forwards the `downlink nack` to integrations and regenerates the downlink. 
+
+Sending a downlink happens only after an uplink for class A devices. If a NACK is received from a class A device, the downlink will automatically be re-encrypted and resent.
+
+On the othed hand, for class B and class C devices, the server behavior for confirmed downlinks is a bit different. The difference arises from the fact that downlinks for class B and class C devices can be sent as part of the downlink slot, without previously receiving an uplink. Uplinks, that carry the ACK/NACK flag, can be sent by these devices at irregular intervals. If the confirmed downlink is sent to the end device, the Network Server will not generate the `downlink ack/nack` message until the next uplink arrives, so the downlink will not be automatically resent.
```

Comment:
```suggestion
On the other hand, for class B and class C devices, the server behavior for confirmed downlinks is a bit different. The difference arises from the fact that downlinks for class B and class C devices can be sent as part of the network initiated downlink slot, without previously receiving an uplink. Uplinks, that carry the ACK/NACK flag, can be sent by these devices at irregular intervals. After the confirmed downlink is transmitted to the end device, the Network Server will not generate the `downlink ack/nack` message until the next uplink arrives, so the downlink will not be automatically resent in case the end device does not reply to the transmission.
```